src/pages/SocialGraph.vue

```diff
@@ -148,32 +211,37 @@ export default {
           );
           return account;
         });
-      return (await Promise.all(promises)).sort(
-        (a, b) => b.totalValue - a.totalValue,
-      );
     },
 
-    async load() {
-      if (!typeMap.has(this.type)) return;
-      const type = typeMap.get(this.type);
+    async fetchPageData(page) {
+      const i = page - 1;
       const params = {
         'pagination.limit': INDEXER_QUERY_LIMIT,
-        'pagination.offset': 0,
+        'pagination.offset': i * INDEXER_QUERY_LIMIT,
         reverse: true,
       };
-      params[type.param] = this.account;
-      this.responseType = type.responseType;
-      let allAccountData = [];
-      let paginationCount;
-      do {
-        // eslint-disable-next-line no-await-in-loop
-        const { data } = await indexerApi.get(`/likechain/likenft/v1/${this.type}`, { params });
-        const accountData = data[type.responseType];
-        allAccountData = allAccountData.concat(accountData);
-        paginationCount = data.pagination.count;
-        params['pagination.offset'] += INDEXER_QUERY_LIMIT;
-      } while (paginationCount === INDEXER_QUERY_LIMIT);
-      this.response = await this.aggregate(allAccountData);
+      params[this.paramField] = this.account;
+      const { data } = await indexerApi.get(`/likechain/likenft/v1/${this.type}`, { params });
+      const accountData = data[this.responseField];
+      const pageData = await Promise.all(this.aggregate(accountData));
+      return pageData;
+    },
+
+    async load() {
+      [
+        this.currentPageData,
+        this.nextPageData,
+      ] = await Promise.all([
+        // eslint-disable-next-line no-console
+        this.fetchPageData(this.currentPage).catch(console.error),
```

Comment:
So nothing will happen if current page fetch throws? Also when I click next page, `currentPageData` is refetched instead of reusing `nextPageData`?

Reply:
Should we display an error on the screen?

Probably use an array/object to cache the `pageData` ever fetch.

Reply:
Error is good, or at least when fail to fetch data should be set empty